src/plugins/syscalls/syscalls.cpp

```diff
@@ -194,6 +199,38 @@ static event_response_t win_cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
         }
     }
 
+    if(!wrapper->ret_trap_set && wsc->ret != VOID){
```

Comment:
You assume the return address for the syscall will always be the same as when it was first called. I don't think that's an assumption that will hold in all cases.